include/sound/intel-nhlt.h

```diff
@@ -75,6 +75,7 @@ struct nhlt_endpoint {
 	u8   direction;
 	u8   virtual_bus_id;
 	struct nhlt_specific_cfg config;
+	struct nhlt_fmt fmt_configs;
```

Comment:
Both 'configs' and 'fmt_configs' now have variable size.... this can't work right?

It is also strange that fmt_configs.fmt_config embeds a copy of "struct nhlt_specific_cfg". This should still work.

Reply:
yes, the variable size can't work, i'll have to figure this one out.

The existing definitions are misleading, they abuse the same structure name for two different purposes.

````
struct nhtl_endpoint {
    ...
    struct nhlt_specific_cfg
    struct nhlt_fmt {
         int fmt_count 
         {
             struct wav_fmt_ext
             struct nhlt_specific_cfg
          }
````